beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -305,6 +305,11 @@ impl<T: BeaconChainTypes> Critical<T> {
         Ok(())
     }
 
+    /// Returns true if the block root is known, without altering the LRU ordering
+    pub fn has_block(&self, block_root: &Hash256) -> bool {
+        self.in_memory.peek(block_root).is_some() || self.store_keys.get(block_root).is_some()
```

Comment:
@ethDreamer is this line right, i.e. should I also check the store keys?

Reply:
I think so, the store keys are referencing the entries in the data availability cache that over overflowed to disk. So it's an extension of the availability cache really. 

Side note - but once we've merged tree states we should consider getting rid of the overflow logic.  It would be a nice complexity reduction.